### PR TITLE
ROC-4310: Refactor out paid in full rule

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link
[ROC-4310](https://tools.hmcts.net/jira/browse/ROC-4310)

### Change description

This PR refactors out paid in full rule, and fixes typo.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@Component
+public class PaidInFullRule {
+
+    public void assertPaidInFull(Claim claim, String claimantId) {
+        requireNonNull(claim, "claim object can not be null");
+        requireNonNull(claimantId, "submitterId object can not be null");
+        if (!claim.getSubmitterId().equals(claimantId)) {
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java]: Suggest flipping `.equals` as this unit knows `claimantId` is not-null at this point, but doesn't know about `claim.getSubmitterId()`

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            throw new ClaimantLinkException(
+                String.format("Claim %s is not linked with claimant %s", claim.getReferenceNumber(), claimantId)
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java]: `format` has been imported

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java (hunk 1 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java]: can not imagine a claim object without a submitter id.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java (hunk 2 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java]: done

**Pull request merged**